### ISSUE TITLE
Python error when opening Acatama Panel

### ISSUE THREAD

**Issue by frnmz** [SMByC/AcATaMa#12]
When opening Acatama panel from toolbar, I got this python error:
```
2022-01-25T17:54:48     WARNING    Traceback (most recent call last):
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/acatama.py", line 127, in run
              self.dockwidget = AcATaMaDockWidget()
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/acatama_dockwidget.py", line 68, in __init__
              self.setup_gui()
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/acatama_dockwidget.py", line 95, in setup_gui
              self.QCBox_ThematicRaster.layerChanged.connect(self.select_thematic_raster)
             TypeError: decorated slot has no signature compatible with layerChanged(QgsMapLayer*)
```
This is my QGIS info:
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" />

QGIS version | 3.22.2-Białowieża | QGIS code branch | Release 3.22
-- | -- | -- | --
Qt version | 5.15.2
Python version | 3.10.0
GDAL/OGR version | 3.3.2
Compiled against PROJ | 8.2.0.%3.%4 | Running against PROJ | 8.2.1
EPSG Registry database version | v10.041 (2021-12-03)
GEOS version | 3.9.2-CAPI-1.14.3
SQLite version | 3.36.0
PDAL version | 2.3.0
PostgreSQL client version | 13.4
SpatiaLite version | 5.0.1
QWT version | 6.1.5
QScintilla2 version | 2.13.0
OS version | Fedora Linux 35 (Workstation Edition)
  |   |   |  
Active Python plugins
CalculateGeometry | 0.6.4
AcATaMa | 19.11.21
SemiAutomaticClassificationPlugin | 7.10.5
MetaSearch | 0.3.5
db_manager | 0.1.20
processing | 2.12.99
grassprovider | 2.12.99
sagaprovider | 2.12.99

</body></html>

This also happens on Windows with the same error message:
```
2022-01-25T17:56:21     WARNING    Traceback (most recent call last):
              File "C:\Users/sakinah.lestari/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\AcATaMa\acatama.py", line 127, in run
              self.dockwidget = AcATaMaDockWidget()
              File "C:\Users/sakinah.lestari/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\AcATaMa\gui\acatama_dockwidget.py", line 68, in __init__
              self.setup_gui()
              File "C:\Users/sakinah.lestari/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\AcATaMa\gui\acatama_dockwidget.py", line 95, in setup_gui
              self.QCBox_ThematicRaster.layerChanged.connect(self.select_thematic_raster)
             TypeError: decorated slot has no signature compatible with layerChanged(QgsMapLayer*)

```

Edit: It does not happened on the LTR version 3.16.16 'Hannover'.

**Comment by XavierCLL** [comment]
Hi @frnmz thanks a lot for your report.

I am aware of this error, it was fixed for the developed version meanwhile I just sent a fixed version for the stable branch: `19.11.21a` please update and try again.

Regards

**Comment by frnmz** [comment]
I can confirm that the panel is fixed now.
But when I choose the raster from panel it gives me python error:
```
2022-01-27T11:03:53     WARNING    Traceback (most recent call last):
              File "/home/lam/.local/share/QGIS/QGIS3/profiles/default/python/plugins/AcATaMa/gui/acatama_dockwidget.py", line 265, in select_thematic_raster
              self.nodata_ThematicRaster.setValue(get_nodata_value(layer))
             TypeError: setValue(self, int): argument 1 has unexpected type 'float'
```

Strangely it's only happened on certain data like this: [error.zip](https://github.com/SMByC/AcATaMa/files/7947493/error.zip)
But not this: [normal.zip](https://github.com/SMByC/AcATaMa/files/7947495/normal.zip)

Both of them is the same classification data I got from Semi-Automatic Classification Plugin processing.
The only difference is the error data is clipped again (even though it's already clipped) to make sure it's the same cell size and extent with another data.

**Comment by XavierCLL** [comment]
Hi @frnmz thanks for your detailed report, I appreciate that a lot, I'm going to check it

**Comment by XavierCLL** [comment]
Fixed in the version 19.11.21b

**Comment by frnmz** [comment]
All problem is fixed now! Thank you so much!
